brave/src/test/java/brave/features/handler/RedactingFinishedSpanHandlerTest.java

```diff
@@ -86,13 +91,36 @@ static String maybeUpdateValue(String value) {
       span.finish();
     }
 
-    assertThat(spans.get(0).tags()).containsExactly(
+    Span finished = spans.take();
+    assertThat(finished.tags()).containsExactly(
       entry("a", "1"),
       // SSN tag was nuked
       entry("c", "3")
     );
-    assertThat(spans.get(0).annotations()).flatExtracting(Annotation::value).containsExactly(
+    assertThat(finished.annotations()).flatExtracting(Annotation::value).containsExactly(
       "SSN=xxx-xx-xxxx"
     );
+
+    // Leak some data by adding a tag using the same context after the span was finished.
+    tracing.tracer().toSpan(span.context()).tag("d", "SSN=912-23-1433");
+    span = null; // Orphans are via GC, to test this, we have to drop any reference to the context
+    blockOnGC();
+
+    // GC only clears the reference to the leaked data. Normal tracer use implicitly handles orphans
+    tracing.tracer().nextSpan().start().finish();
+
+    Span leaked = spans.take();
+    assertThat(leaked.tags()).containsExactly(
+      // SSN tag was nuked
+      entry("d", "SSN=xxx-xx-xxxx")
+    );
+    assertThat(leaked.annotations()).flatExtracting(Annotation::value).containsExactly(
```

Comment:
Maybe use the new method

Reply:
agreed

Reply:
the method is on mutablespan not zipkin2.span :P lemme see if it makes sense to refactor the test..

Reply:
meh creates some busy work to do that as most assertions work more naturally against zipkin2.Span